### PR TITLE
dialects: (dmp)A bunch of changes that came up when integrating xDSL directly into devito.

### PR DESCRIPTION
**Pull request by AntonLydike**
Note: this will break the current benchmarking setup for devito.

### CODE REVIEW

**Comment by AntonLydike** [comment]
Merging will coincide with a PR on the devito PR so that we don't break benchmarking. Just FYI.

**Review by georgebisbas — APPROVED** [review]
lovely, thanks

**Review by georgebisbas — APPROVED** [review]

File: .github/workflows/update_xdsl_pyodide_build.py [hunk 1]
```diff
@@ -3,10 +3,11 @@
...
 # Takes the .yaml file and the xDSL directory as arguments
 
-import yaml  # type: ignore
-import sys
-import os
 import hashlib
+import os
+import sys
+
+import yaml  # type: ignore
```
Comment (superlopuh) [inline .github/workflows/update_xdsl_pyodide_build.py]: why type ignore? Can you replace with specific ignore as opposed to catch-all, please?

Comment (AntonLydike) [reply .github/workflows/update_xdsl_pyodide_build.py]: This change is not part of the PR, you see the diff from merging main into this branch

**Pull request merged**